### PR TITLE
Added partial support for Element.scrollIntoView()

### PR DESCRIPTION
**Pull request by andrei0x309**
Since there isn't a way now to programmatically scroll to an element, I thought it would be nice if there was some basic support for that.

I've seen some issues like [issue-1009](https://github.com/ampproject/worker-dom/issues/1009)  that can be resolved if `worker-dom` supports `scrollIntoView()`, I only added basic support(no support for method params) since it required less code, also the 'smooth' option today is supported by using CSS in most browsers, and those that don't support CSS smooth, they don't support `scrollIntoView()` with smooth as an option.

I couldn't write a test since I believe `Window.pageY` was necessary and is not supported by `worker-dom`, but I wrote a demo instead of a test. 

Thanks, and I appreciate any feedback.

### CODE REVIEW

**Comment by samouri** [comment]
@andrei0x309, thank you so much for the PR! At initial glance this looks great. I'll give it a more thorough review later today.

One quick note: we usually write unit tests that ensure messages passed between main-thread and worker-thread are as expected, as opposed to making them integration tests (which could involve checking `Window.pageY`).

Therefore potential tests you could write:
1. Ensure worker-thread sends the expected message to the main-thread when Element.scrollIntoView is called.
2. Ensure main-thread calls the expected DOM API when receiving the correct message from the worker-thread


Here's an example of a test expecting mutation events to be sent from the worker-thread:
https://github.com/ampproject/worker-dom/blob/f5da1e6231d552618290ae07cdfd4ac547b3ff12/src/test/mutation-transfer/amp.test.ts#L42-L57

**Comment by andrei0x309** [comment]
@samouri Thanks, for the reply, 

I added the transfer mutation test for scrollIntoView().

File: src/transfer/TransferrableMutation.ts [hunk 1]
```diff
@@ -44,6 +45,7 @@ export const isUserVisibleMutation = (type: TransferrableMutationType): boolean
     case TransferrableMutationType.STORAGE:
     case TransferrableMutationType.OFFSCREEN_CANVAS_INSTANCE:
     case TransferrableMutationType.FUNCTION_CALL:
+    case TransferrableMutationType.SCROLL_INTO_VIEW:
```
Comment (samouri) [inline src/transfer/TransferrableMutation.ts:48]: I know the comment describes user-visible changes to _DOM_, but I actually think the intention is any user-visible change (which would include this). Can you delete this line s.t. it returns true
```suggestion
```

File: src/test/mutation-transfer/scrollIntoView.test.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+ import anyTest, { TestInterface } from 'ava';
+ import { Document } from '../../worker-thread/dom/Document';
+ import { TransferrableMutationType } from '../../transfer/TransferrableMutation';
+ import { createTestingDocument } from '../DocumentCreation';
+ import { expectMutations } from '../Emitter';
+ import { Element } from '../../worker-thread/dom/Element';
+import { TransferrableKeys } from '../../transfer/TransferrableKeys';
```
Comment (samouri) [inline src/test/mutation-transfer/scrollIntoView.test.ts:23]: nit: did all of the prepush checks run? we normally run `prettier` over the codebase, and the whitespace here doesn't look right.

File: demo/scroll-into-view/scroll-into-view.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+document.getElementById('scroll-button').addEventListener('click', () => {
+  document.getElementById('scroll-element').scrollIntoView();
+  });
```
Comment (samouri) [inline demo/scroll-into-view/scroll-into-view.js:3]: nit: whitespace here doesn't look right. 
```suggestion
});
```

File: demo/scroll-into-view/scroll-into-view.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+document.getElementById('scroll-button-up').addEventListener('click', () => {
+  document.getElementById('scroll-button').scrollIntoView();
+  });
```
Comment (samouri) [inline demo/scroll-into-view/scroll-into-view.js:7]:
```suggestion
});  
```

File: demo/scroll-into-view/index.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    <link href="/demo.css" rel="stylesheet" />
+    <script src="/dist/main.mjs" type="module"></script>
+    <script src="/dist/main.js" nomodule defer></script>
+    <!-- This comment block is intended to make it easier to test both the script module and nomodule path -->
+    <!-- Comment either block to enable module/nomodule or disable it. -->
+    <!-- <script src="/dist/main.js" defer></script> -->
+  </head>
+  <body>
+    <div src="/scroll-into-view/scroll-into-view.js" id="upgrade-me">
+      <p>Demo of <b>element.scrrollIntoView()</b></p>
```
Comment (samouri) [inline demo/scroll-into-view/index.html:16]:
```suggestion
      <p>Demo of <b>element.scrollIntoView()</b></p>
```

**Review by samouri — CHANGES_REQUESTED** [review]
Excellent PR :). Only one blocking request (to ensure it is considered user visible).
Otherwise LGTM.

As an action item on my side I'm going to make a new command, `yarn format` for automatically running prettier on files

File: src/transfer/TransferrableMutation.ts (hunk 1 above)
Comment (andrei0x309) [reply src/transfer/TransferrableMutation.ts]: Yeah, you're completely right I only thought of DOM and not considered the scroll changing position as a user-visible change which it actually is.

File: src/test/mutation-transfer/scrollIntoView.test.ts (hunk 2 above)
Comment (andrei0x309) [reply src/test/mutation-transfer/scrollIntoView.test.ts:23]: This time I didn't. 
Sorry for that.

**Review by samouri — APPROVED** [review]

**Pull request merged**